src/unix.rs

```diff
@@ -21,13 +22,24 @@ pub struct Acquired {
 }
 
 impl Client {
-    pub fn new(limit: usize) -> io::Result<Client> {
+    pub fn new(mut limit: usize) -> io::Result<Client> {
+        if limit > PIPE_BUF {
+            return Err(io::Error::new(io::ErrorKind::Other,  "Can't have limit larger than 4096 since the pipe would block forever when writing to it"));
+        }
```

Comment:
Personally I still don't think this should be done. I'm pretty sure that the limit of a pipe is a runtime-configurable thing in the kernel and not a static constant someone can write down in libc.

Reply:
> I'm pretty sure that the limit of a pipe is a runtime-configurable thing in the kernel and not a static constant someone can write down in libc.

You are right, this `libc::PIPE_BUF` only specifies the minimum size of pipe.

I would switch non-blocking then, but I would have to remove `O_NONBLOCK` before returning from `new`.

Reply:
Done.